src/tz.cpp

```diff
@@ -1447,7 +1447,11 @@ find_previous_rule(const Rule* r, date::year y)
     if (y == r->starting_year())
     {
         if (r == &rules.front() || r->name() != r[-1].name())
-            std::terminate();  // never called with first rule
+        {
+            throw std::runtime_error("find_previous_rule called but no previous rule");
+            Rule *noRule;
+            return {noRule, date::year::min()};
```

Comment:
Why not `return {nullptr, date::year::min()};` ?

Reply:
If we get here, it's a logic error in tz.cpp.  There's no way to continue executing with that level of uncertainty of the program state.  If there is a way for the client to get here without it being a logic error in tz.cpp, I'd rather diagnose it earlier.